### PR TITLE
Run the built supervisor as part of docker-compose tests

### PR DESCRIPTION
**Pull request by pipex**
This allows to test that the supervisor build actually runs and opens up the possibility of running more exhaustive API tests against a working supervisor.

Change-type: patch

### CODE REVIEW

**Comment by cywang117** [comment]
@balena-ci rebase

File: test/integration/supervisor.spec.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import * as request from 'supertest';
+
+const BALENA_SUPERVISOR_ADDRESS =
+	process.env.BALENA_SUPERVISOR_ADDRESS || 'http://balena-supervisor:48484';
+
+describe('supervisor app', () => {
```
Comment (cywang117) [inline test/integration/supervisor.spec.ts:6]: Could we consider removing `test/legacy/18-startup.spec.ts` since we have this?

Comment (pipex) [reply test/integration/supervisor.spec.ts:6]: yep, good call

Comment (pipex) [reply test/integration/supervisor.spec.ts:6]: done

**Review by cywang117 — APPROVED** [review]

**Pull request merged**